app/components/build/AnnotationsList.js

```diff
@@ -41,10 +57,40 @@ class AnnnotationsList extends React.Component<Props> {
     });
 
     return (
-      <div>{annotations}</div>
+      <div>
+        {annotations}
+        {this.renderShowMore()}
+      </div>
     );
   }
 
+  renderShowMore() {
+    if (this.props.build.annotations.pageInfo.hasNextPage) {
```

Comment:
You don’t need to check for hasNextPage here; `ShowMoreFooter` can check that for you:

https://github.com/buildkite/frontend/blob/e8d97f5d5c9ed2faba3f8924894b02d1c59e9016/app/components/shared/ShowMoreFooter/index.js#L26-L38

Reply:
Oh awesome!